### PR TITLE
chore(apigateway): improve docs and default value settings for identity source in authorizer

### PR DESCRIPTION
**Pull request by go-to-k**
Addressed the following points about Authorizer's `identitySource` property.

- Reference link is broken.
  - now: https://docs.aws.amazon.com/apigateway/api-reference/link-relation/authorizer-create/#identitySource
  - new: https://docs.aws.amazon.com/apigateway/latest/api/API_CreateAuthorizer.html#apigw-CreateAuthorizer-request-identitySource
- One explanation was confusing, so I enclosed it in quotes.
  - now: ```this should be `method.request.header.Authorizer` where Authorizer is the header containing the bearer token.```
  - new: ```this should be `method.request.header.Authorizer` where `Authorizer` is the header containing the bearer token.```
- Not using the static method written in the doc to set default values when a prop is not specified.
  - now: `'method.request.header.Authorization'`
  - new: `IdentitySource.header('Authorization')`

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-apigateway/lib/authorizers/cognito.ts [hunk 1]
```diff
@@ -78,7 +80,7 @@ export class CognitoUserPoolsAuthorizer extends Authorizer implements IAuthorize
       type: 'COGNITO_USER_POOLS',
       providerArns: props.cognitoUserPools.map(userPool => userPool.userPoolArn),
       authorizerResultTtlInSeconds: props.resultsCacheTtl?.toSeconds(),
-      identitySource: props.identitySource || 'method.request.header.Authorization',
+      identitySource: props.identitySource || IdentitySource.header('Authorization'),
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-apigateway/lib/authorizers/cognito.ts:83]: The change caused by this is guaranteed by [this test](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-apigateway/test/authorizers/cognito.test.ts#L27).

File: packages/aws-cdk-lib/aws-apigateway/lib/authorizers/lambda.ts [hunk 2]
```diff
@@ -216,7 +218,7 @@ export class TokenAuthorizer extends LambdaAuthorizer {
       authorizerUri: lambdaAuthorizerArn(props.handler),
       authorizerCredentials: props.assumeRole?.roleArn,
       authorizerResultTtlInSeconds: props.resultsCacheTtl?.toSeconds() ?? Duration.minutes(5).toSeconds(),
-      identitySource: props.identitySource || 'method.request.header.Authorization',
+      identitySource: props.identitySource || IdentitySource.header('Authorization'),
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-apigateway/lib/authorizers/lambda.ts:221]: The change caused by this is guaranteed by [this test](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-apigateway/test/authorizers/lambda.test.ts#L30).

File: packages/aws-cdk-lib/aws-apigateway/lib/authorizers/cognito.ts [hunk 3]
```diff
@@ -33,8 +34,9 @@ export interface CognitoUserPoolsAuthorizerProps {
 
   /**
    * The request header mapping expression for the bearer token. This is typically passed as part of the header, in which case
-   * this should be `method.request.header.Authorizer` where Authorizer is the header containing the bearer token.
-   * @see https://docs.aws.amazon.com/apigateway/api-reference/link-relation/authorizer-create/#identitySource
+   * this should be `method.request.header.Authorizer` where `Authorizer` is the header containing the bearer token.
+   *
+   * @see https://docs.aws.amazon.com/apigateway/latest/api/API_CreateAuthorizer.html#apigw-CreateAuthorizer-request-identitySource
    * @default `IdentitySource.header('Authorization')`
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-apigateway/lib/authorizers/cognito.ts:40]: The default value setting process has been changed to fit here.

File: packages/aws-cdk-lib/aws-apigateway/lib/authorizers/lambda.ts [hunk 4]
```diff
@@ -182,8 +183,9 @@ export interface TokenAuthorizerProps extends LambdaAuthorizerProps {
 
   /**
    * The request header mapping expression for the bearer token. This is typically passed as part of the header, in which case
-   * this should be `method.request.header.Authorizer` where Authorizer is the header containing the bearer token.
-   * @see https://docs.aws.amazon.com/apigateway/api-reference/link-relation/authorizer-create/#identitySource
+   * this should be `method.request.header.Authorizer` where `Authorizer` is the header containing the bearer token.
+   *
+   * @see https://docs.aws.amazon.com/apigateway/latest/api/API_CreateAuthorizer.html#apigw-CreateAuthorizer-request-identitySource
    * @default `IdentitySource.header('Authorization')`
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-apigateway/lib/authorizers/lambda.ts:189]: The default value setting process has been changed to fit here.

**Review by lpizzinidev — APPROVED** [review]
Looks good to me 👍

**Pull request merged**